coalib/results/Diff.py

```diff
@@ -70,6 +72,57 @@ def from_string_arrays(cls, file_array_1, file_array_2, rename=False):
         return result
 
     @classmethod
+    def from_unified_diff(cls, unified_diff, original_file):
+        """
+        Creates a ``Diff`` object from given unified diff.
+
+        If this ``Diff`` is applied to the original file, the corrected
+        file will be created.
+
+        :param unified_diff:  Unified diff string.
+        :param original_file: The contents of the original file
+                              (line-splitted).
+        """
+        patch_set = PatchSet(unified_diff.splitlines())
+
+        diff = Diff(original_file)
+
+        for file in patch_set:
+            for hunk in file:
+                file_line = hunk.source_start
+                i = 0
```

Comment:
as you only use `i` to control the loop and only increment by a fixed amount, you could try to use iterators so you don't have to do `i += 1`, like
```python
hunk_iterator = iter(hunk)
line = next(hunk_iterator)
```